src/TServer.cpp

```diff
@@ -459,9 +464,45 @@ void TServer::HandlePosition(TClient& c, const std::string& Packet) {
     if (MaybePidVid) {
         int PID = -1;
         int VID = -1;
-        // FIXME: check that the VID and PID are valid, so that we don't waste memory
         std::tie(PID, VID) = MaybePidVid.value();
+        if (PID == c.GetID()) {
+            c.SetCarPosition(VID, Data);
+            return true;
+        }
+    }
+    return false;
+}
+
+bool TServer::HandleVehicleUpdate(TClient& c, const std::string& Packet) {
+    if (Packet.size() < 3) {
+        // invalid packet
+        return false;
+    }
+    // (Vi/We/Yl):serverVehicleID:data
+    // (Vi/We/Yl):serverVehicleID:data
+    std::string withoutCode = Packet.substr(3);
+    auto NameDataSep = withoutCode.find(':', 2);
+    if (NameDataSep == std::string::npos || NameDataSep < 2) {
+        // invalid packet
+        return false;
+    }
+    // FIXME: ensure that -2 does what it should... it seems weird.
```

Comment:
I agree this seems weird, and I'd ask to add explicit length checks before every single substr() call - if they're out of bounds, the server dies violently, so we need to be very careful (double, triple check we're not out of bounds).

Reply:
that chunk is a copy of the function above, but i agree it could do with more safeties